src/utils/utils.ts

```diff
@@ -9,9 +9,9 @@ export function trimIpfsHash(ipfsHash: string): string {
 
 export function extractNFT(nftResponse: INFTResponse): IMetadata {
     const nft: IMetadata = {
-        name: nftResponse.metadata.name,
-        description: nftResponse.metadata.description,
-        video: nftResponse.metadata.video,
+        name: nftResponse?.metadata?.name || '',
```

Comment:
Again, if the name can be undefined (and it really can be), you should return undefined in the other comment, and not '-'